go/pkg/sysdb/metastore/db/dao/collection.go

```diff
@@ -27,10 +27,16 @@ func (s *collectionDb) DeleteAll() error {
 
 func (s *collectionDb) GetCollectionEntry(collectionID *string, databaseName *string) (*dbmodel.Collection, error) {
 	var collections []*dbmodel.Collection
-	err := s.db.Table("collections").
+	query := s.db.Table("collections").
 		Select("collections.id, collections.name, collections.database_id, collections.is_deleted, databases.name, databases.tenant_id").
 		Joins("INNER JOIN databases ON collections.database_id = databases.id").
-		Where("collections.id = ? AND databases.name = ?", collectionID, databaseName).Find(&collections).Error
+		Where("collections.id = ?", collectionID)
+
+	if databaseName != nil && *databaseName != "" {
```

Comment:
Why are both these conditions checked for?  As a reader I'd ask why this is and dig into it.  A comment here would be helpful.